adcm_aio_client/core/objects/cm.py

```diff
@@ -1,31 +1,42 @@
 from functools import cached_property
-from typing import Iterable, Self
+from typing import Any, Iterable, Self
 import asyncio
 
 from asyncstdlib.functools import cached_property as async_cached_property
 
 from adcm_aio_client.core.errors import NotFoundError, OperationError, ResponseError
 from adcm_aio_client.core.objects._accessors import (
+    NonPaginatedChildAccessor,
     PaginatedAccessor,
     PaginatedChildAccessor,
 )
 from adcm_aio_client.core.objects._base import InteractiveChildObject, InteractiveObject, RootInteractiveObject
 from adcm_aio_client.core.objects._common import (
     Deletable,
     WithActionHostGroups,
-    WithActions,
     WithConfig,
     WithConfigGroups,
     WithStatus,
     WithUpgrades,
 )
 from adcm_aio_client.core.objects._imports import ClusterImports
 from adcm_aio_client.core.objects._mapping import ClusterMapping
-from adcm_aio_client.core.types import ADCMEntityStatus, Endpoint
+from adcm_aio_client.core.types import ADCMEntityStatus, AwareOfOwnPath, Endpoint, Requester, WithRequester
 
 type Filter = object  # TODO: implement
 
 
+# TODO: Avoiding circular imports. Think about right place for such mixins.
+class WithActions(WithRequester, AwareOfOwnPath):
+    @cached_property
+    def actions(self: Self) -> "ActionsAccessor":
+        return ActionsAccessor(
+            parent=self,  # pyright: ignore [reportArgumentType]
```

Comment:
why's ignore? better rule that out now

looks like too much restrictions on parent or too little inheritance on WithActions

Reply:
refactored